pkg/inject/inject.go

```diff
@@ -63,53 +63,59 @@ var injectableKinds = []string{
 	k8s.StatefulSet,
 }
 
-// objMeta provides a generic struct to parse the names of Kubernetes objects
-type objMeta struct {
-	*metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
-}
-
 // ResourceConfig contains the parsed information for a given workload
 type ResourceConfig struct {
 	globalConfig           *config.Global
 	proxyConfig            *config.Proxy
 	nsAnnotations          map[string]string
-	meta                   metav1.TypeMeta
-	obj                    runtime.Object
-	workLoadMeta           *metav1.ObjectMeta
-	podMeta                objMeta
-	podLabels              map[string]string
-	podSpec                *v1.PodSpec
 	destinationDNSOverride string
 	proxyOutboundCapacity  map[string]uint
+
+	workload struct {
+		obj      runtime.Object
+		meta     *metav1.ObjectMeta
+		metaType metav1.TypeMeta
+	}
+
+	pod struct {
+		// Meta is the pod's metadata. It's exported so that the YAML marshaling
+		// will work in the ParseMeta() function.
+		Meta   *metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
Oh.. having the `Meta` inside `pod` plays the same role `objMeta` did for parsing the yaml, but this is much nicer :100: